SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Shared.Targets/CrashesUtils.cs

```diff
@@ -9,41 +8,37 @@ internal static class CrashesUtils
     {
         internal static byte[] SerializeException(Exception exception)
         {
-            var ms = new MemoryStream();
-            var formatter = new BinaryFormatter();
-
             try
             {
-                formatter.Serialize(ms, exception);
+                using (var memoryStream = new MemoryStream())
+                {
+                    var formatter = new BinaryFormatter();
+                    formatter.Serialize(memoryStream, exception);
+                    return memoryStream.ToArray();
+                }
             }
-            catch (SerializationException e)
+            catch (Exception e)
             {
                 AppCenterLog.Warn(Crashes.LogTag, "Failed to serialize exception for client side inspection", e);
-                ms = new MemoryStream();
-                formatter.Serialize(ms, e);
             }
-
-            return ms.ToArray();
+            return null;
         }
 
         internal static Exception DeserializeException(byte[] exceptionBytes)
         {
-            var ms = new MemoryStream(exceptionBytes);
-            var formatter = new BinaryFormatter();
-
-            Exception deserializedException;
-
             try
             {
-               deserializedException = formatter.Deserialize(ms) as Exception;
+                using (var memoryStream = new MemoryStream(exceptionBytes))
+                {
+                    var formatter = new BinaryFormatter();
+                    return formatter.Deserialize(memoryStream) as Exception;
+                }
             }
-            catch(SerializationException e)
+            catch (Exception e)
```

Comment:
Do you know what other type is thrown? Should avoid catching knives like outofmemoryexception or similar (thinking about the when(e.IsNoFatal) extension we have in our backend code). For serialize we already crashed so I am not concerned, but deserialize is called in normal app running scenario.

Reply:
Please note, if it's crashed on deserialize we'll not send original crash since it happens on `OnBeforeSending` callback for example. But I agree with you about fatal ones.